doc/design/design_submit_job.md

```diff
@@ -0,0 +1,60 @@
+# Submit SQLFLow SQL Asynchronously
+
+## Abstract
+
+In the current system, SQLFlow client connects the SQLFlow server with long links. the SQLFlow client sends
+a request which contains the user-typed SQL and hung until the SQLFlow server finishes executing the SQL.
+Sometimes, it would cost a long time to execute the SQL and lead to gRPC time out.
+The purpose of this proposal is to introduce a way to submit the SQLFlow SQL asynchronously to avoid the
+time out issue.
```

Comment:
Maybe mentioning all the possible solutions. For example

-----

## Possible solution
 
1. Increase the default timeout of RPC calls.
1. Communicate via multiple RPC calls. And we maintain the state:
    1. On the client-side: the client starts a service after submitting a SQL, and the server pushes the logs to the client's service.
    1. On the server-side: the client submits a SQL and receives a token. The client poll the server periodically using the token. (Described in this design)
    1. On a third party service: the client submits a SQL and receives a token. The server pushes the logs to a third-party service like etcd or Kafka. The client fetches the result from the third party service using the token.

Each solution has its pros and cons. We choose ... because ...